File: src/Listas3/pratica/Questao03.java
package Listas3.pratica;

import java.util.Scanner;

class DepositoInvalidoException extends RuntimeException {
    public DepositoInvalidoException(String mensagemErro) {
        super(mensagemErro);
    }
}

class Conta {
    private int agencia;
    private int numero;
    private double saldo;
    private static final double LIMITE_DEPOSITO = 10000;

    public Conta(int agencia, int numero) {
        this.agencia = agencia;
        this.numero = numero;
        this.saldo = 0.0;
    }

    public double getSaldo() {
        return saldo;
    }

    public static double getLimiteDeposito() {
        return LIMITE_DEPOSITO;
    }

    public void depositaPersonalizado(double valor) {
        if (valor <= 0) {
            throw new DepositoInvalidoException("Valor inválido para depósito");
        } else if (valor > 10000) {
            throw new DepositoInvalidoException("Valor acima do limite permitido de 10000.00");
        } else {
            this.saldo = valor;
        }
    }
}

public class Questao03 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int agencia = sc.nextInt();
        int numero = sc.nextInt();
        double valorDeposito = 0;

        Conta cont1 = new Conta(agencia, numero);
        try {
            valorDeposito = sc.nextDouble();
            cont1.depositaPersonalizado(valorDeposito);
            System.out.println("Depósito realizado com sucesso.");
        } catch (DepositoInvalidoException e) {
            System.out.println("Erro: " + e.getMessage());
        } finally {
            double saldo = cont1.getSaldo();
            if (valorDeposito <= 0 || valorDeposito > 10000) {
                System.out.printf("Valor %.2f incorreto. Saldo atual: %.2f\n", valorDeposito, saldo);
            } else {
                System.out.printf("Valor %.2f depositado na conta. Novo saldo: %.2f\n", valorDeposito, saldo);
            }
        }
    }
}

File: Listas C++/Lista 3/questao03.cpp
#include <iostream>
#include <string>
#include <stdexcept>
#include <iomanip>
#include <locale>

class DepositoInvalidoException : public std::runtime_error {
public:
    DepositoInvalidoException(const std::string& mensagemErro) : std::runtime_error(mensagemErro) {}
};

class Conta {
private:
    int agencia;
    int numero;
    double saldo;
    static constexpr double LIMITE_DEPOSITO = 10000.0;

public:
    Conta(int agencia, int numero) {
        this->agencia = agencia;
        this->numero = numero;
        this->saldo = 0.0;
    }

    double getSaldo() const {
        return saldo;
    }

    static double getLimiteDeposito() {
        return LIMITE_DEPOSITO;
    }

    void depositaPersonalizado(double valor) {
        if (valor <= 0) {
            throw DepositoInvalidoException("Valor inválido para depósito");
        } else if (valor > LIMITE_DEPOSITO) {
            throw DepositoInvalidoException("Valor acima do limite permitido de 10000.00");
        } else {
            this->saldo = valor;
        }
    }
};

int main() {
    std::cout.imbue(std::locale("C"));

    int agencia, numero;
    std::cin >> agencia >> numero;

    double valorDeposito = 0;
    Conta cont1(agencia, numero);

    try {
        std::cin >> valorDeposito;
        cont1.depositaPersonalizado(valorDeposito);
        std::cout << "Depósito realizado com sucesso." << std::endl;
    } catch (const DepositoInvalidoException& e) {
        std::cout << "Erro: " << e.what() << std::endl;
    }

    double saldo = cont1.getSaldo();
    
    std::cout << std::fixed << std::setprecision(2);

    if (valorDeposito <= 0 || valorDeposito > 10000) {
        std::cout << "Valor " << valorDeposito << " incorreto. Saldo atual: " << saldo << std::endl;
    } else {
        std::cout << "Valor " << valorDeposito << " depositado na conta. Novo saldo: " << saldo << std::endl;
    }

    return 0;
}